cloud/google/gce_net.py

```diff
@@ -110,6 +110,35 @@
     required: false
     default: null
     aliases: []
+  mode:
+    version_added: "2.2"
+    description:
+      - network mode supporting subnets introduced into Google Cloud
+    required: false
+    default: "legacy"
+    choices: ["legacy", "auto", "custom"]
+    aliases: []
+  subnet_name:
+    version_added: "2.2"
+    description:
+      - name of subnet to create
+    required: false
+    default: null
+    aliases: []
+  subnet_region:
+    version_added: "2.2"
+    description:
+      - region of subnet to create
+    required: false
+    default: null
+    aliases: []
+  subnet_desc:
+    version_added: "2.2"
+    description:
+      - description of subnet to create
+    required: false
+    default: null
+    aliases: []
```

Comment:
Can you please add an example of using the subnet/mode parameters, and a link to the GCE docs on the mode terms. legacy/auto/custom aren't very transparent terms.